### ISSUE TITLE
Windows RailsInstaller for Rails 4.2 and Ruby 2.2 still broken?

### ISSUE THREAD

**Issue by edactic** [railsinstaller/railsinstaller-windows#81]
I just downloaded and tried RailsInstaller for Rails 4.2 and Ruby 2.2 from installRails.com.  Things appear to install correctly but the paths to the executables are still apparently corrupt.  No changes were made to the Windows PATH configuration, and so nothing works the way it should.   Can't run ruby, rails, or git, to name the top three.  (Well, not completely true.  I can run Ruby if I manually go to C:\RailsInstaller\Ruby2.2.0\bin).    When I look at paths in some batch files, I see things like: "C:\Users\emachnic\GitRepos\..." as if hard-coded paths were randomly left in place by the developer.     At least 2 months have passed since a bunch of threads were posted about problems like this and apparently still no fixes.   Quite a disappointment and not a good way to convert a developer to Rails as a viable platform.

**Comment by margaretcsmith** [comment]
Hi,
I still can't install Rails on my Windows machine. Any news on when the fix will be ready?
Thanks,
M

**Comment by Seally** [comment]
Just change the paths in those files to the correct path. Going by the pattern, a simple replace all by a text editor that can see the entire tree (e.g. Sublime, Atom, most IDEs) should be able to solve most of these issues. This has been reported (and closed) already:
https://github.com/railsinstaller/railsinstaller-windows/issues/73

There are a lot of path mistakes in RailsInstaller though (I didn't install the Rails Git and found the easiest way to fix the issue was to create an NTFS junction with mklink to my existing Git installation.)

**Comment by margaretcsmith** [comment]
Thanks very much. It worked.

**Comment by josephgodwinkimani** [comment]
Just to make it clear - the files to edit are all gem BAT files in `%HOMEDRIVE%\RailsInstaller\Ruby2.2.0\bin` or wherever your RailsInstaller installation is situated `cd` down to bin directory of your Ruby installation: _bundle, bundler, compass, erb, erubis, gem, irb, rails, sass, tilt, sqlite3_ruby, thor, nokogiri._ Do not tamper with `publickey.bat` unless you know what you are doing.

**Comment by margaretcsmith** [comment]
Hi,
Apparently, I had some syntax errors when I tried to correct the various files involved. I finally found an easy way to correct the issue.  In Powershell, I execute this command to overwrite the previous installation:  "gem install rails" .  Now when I  enter "rails -v", it finds the path. Voila!
Thanks for everyone's help.

**Comment by sebesbal** [comment]
It's ridiculous that bug is still there... (hard coded paths to the developer's machine...???) Give me back my 20 minutes!

**Comment by mchuang484** [comment]
Thanks margaretcsmith.  "gem install rails" solved my installation issue.

**Comment by ashmaroli** [comment]
After successful installation of Railsinstaller, open the bundled version of command-prompt and simply run `gem update --system` to update rubygems version.
Then run `gem update --no-document` to update all installed gems in PATH. This will update associated executables, .bat files and .cmd files in the /bin folder

**Comment by wpostma** [comment]
How can I contribute and get the out of box installation experience fixed? Some developer who knows, please tell me.

This is silly that it's been broken from March (see google groups) to October of 2016. That's 8 months.

I'm new to ruby but I know a bad batch file when I see one.

Maybe the MR should be "run gem update --system" and "gem update --no-document" when the install completes, and maybe the MR should be to drop these broken and outdated ruby binaries from ruby installer and actually just have the ruby installer do "gem install rails"

**Comment by gptura** [comment]
Yup confirmed. Did a fresh install just now and its still broken. I did try changing the paths of some batch files as what the others have mentioned but did not work for me. 

![image](https://cloud.githubusercontent.com/assets/12121694/19915613/af49dd20-a0ee-11e6-9d99-117413024ba0.png)

![image](https://cloud.githubusercontent.com/assets/12121694/19915638/c9ceac8e-a0ee-11e6-8435-4bbcbdc66289.png)

Will probably continue using a cloud IDE for now. I hope the developers would see and resolve this pretty soon. Thanks!

**Comment by gptura** [comment]
My issues in installing rails got resolved using the guide on this website.

http://installrails.com/

**Comment by ashmaroli** [comment]
alternatively, simply run the following to have the latest versions of all installed gems:
```
gem update --system && gem update --no-document
```

**Comment by williaba** [comment]
@ashmaroli thank you so much, "gem update --no-document" worked for me!

**Comment by MichelleBruegger** [comment]
The bug with the wrong paths in different batchfiles has not been corrected with Railsinstaller-3.3.0.exe

First step was:  install until the "Completing the RailsInstaller Setup Wizard" was shown. Don't click on "Finish" yet.

Do what @margaretcsmith commented on 25 Jun 2016.
Then do what @ashmaroli commented on 14 Aug 2016 

After that I changed config_check.rb (located on C:\RailsInstaller\scripts) line 78 and line 82, because they referred to a wrong railsinstaller_path (2.0.0 instead of Ruby2.3.0")

Then I clicked on "Finish" on the RailsInstaller-Setup with activated "Configure git and ssh when installation has completed."

And... it worked 👍

**Comment by emachnic** [comment]
Please try with the latest version.
